packages/fxa-auth-server/fxa-oauth-server/lib/routes/destroy.js

```diff
@@ -2,21 +2,28 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+const crypto = require('crypto');
 const Joi = require('joi');
 
 const AppError = require('../error');
 const db = require('../db');
 const encrypt = require('../encrypt');
 const validators = require('../validators');
 const { getTokenId } = require('../token');
+const { authenticateClient, clientAuthValidators } = require('../client');
 
 /*jshint camelcase: false*/
 
 module.exports = {
   validate: {
+    headers: clientAuthValidators.headers,
     payload: Joi.object()
       .keys({
-        client_secret: Joi.string().allow(''),
+        client_id: clientAuthValidators.clientId.optional(),
+        // For historical reasons, we accept and ignore a client_secret if one
```

Comment:
Do we know whether this still happens and if so, which services are sending a client_secret without a client_id?

Reply:
I do not, and I suspect none are. I'll add a log line while I'm in here so we can evaluate.